lib/src/core/better_player_controller.dart

```diff
@@ -270,6 +270,38 @@ class BetterPlayerController {
     setTrack(BetterPlayerAsmsTrack.defaultTrack());
   }
 
+  void setupFromOtherController(BetterPlayerController otherBetterPlayerController) {
+    final betterPlayerDataSource = otherBetterPlayerController.betterPlayerDataSource;
+    if (betterPlayerDataSource == null) return;
+
+    postEvent(BetterPlayerEvent(BetterPlayerEventType.setupDataSource, parameters: <String, dynamic>{
+      _dataSourceParameter: betterPlayerDataSource,
+    }));
+    _postControllerEvent(BetterPlayerControllerEvent.setupDataSource);
+    _betterPlayerDataSource = betterPlayerDataSource;
+    _hasCurrentDataSourceStarted = otherBetterPlayerController.hasCurrentDataSourceStarted;
+    _hasCurrentDataSourceInitialized = otherBetterPlayerController._hasCurrentDataSourceInitialized;
+    _betterPlayerAsmsTracks = otherBetterPlayerController.betterPlayerAsmsTracks;
+    _betterPlayerAsmsTrack = otherBetterPlayerController.betterPlayerAsmsTrack;
+    _betterPlayerSubtitlesSourceList.clear();
+    _betterPlayerSubtitlesSourceList.addAll(otherBetterPlayerController.betterPlayerSubtitlesSourceList);
+    _betterPlayerAsmsAudioTracks = otherBetterPlayerController.betterPlayerAsmsAudioTracks;
+    _betterPlayerAsmsAudioTrack = otherBetterPlayerController.betterPlayerAsmsAudioTrack;
+
+    videoPlayerController = otherBetterPlayerController.videoPlayerController;
+    otherBetterPlayerController.detachVideoPlayerController();
+    otherBetterPlayerController.dispose(forceDispose: true);
```

Comment:
should we move this to `detachVideoPlayerController()`?

Reply:
refer logic check in `dispose` function.

Reply:
> should we move this to detachVideoPlayerController()?

chuyển vào thì nó không còn giữ đúng chức năng của func này nữa rồi anh ạ,